Code/GraphMol/Substruct/SubstructUtils.cpp

```diff
@@ -246,11 +246,14 @@ std::vector<MatchVectType> sortMatchesByDegreeOfCoreSubstitution(
 }
 
 bool isAtomTerminalRGroupOrQueryHydrogen(const Atom *atom) {
-  return atom->getDegree() == 1 &&
-         (atom->getAtomicNum() == 0 ||
-          (atom->hasQuery() &&
-           describeQuery(atom).find("AtomAtomicNum 1 = val") !=
-               std::string::npos));
+  auto isDummyAtom = [](const Atom &atom) {
+    return (!atom.hasQuery() && atom.getAtomicNum() == 0) ||
+           (atom.hasQuery() && atom.getQuery()->getDescription() == "AtomNull");
+  };
```

Comment:
please use the version of this in QueryOps.h (assuming you make that suggested change)

Reply:
Done.